### PR TITLE
tests rayci with rayci

### PR DESCRIPTION
**Pull request by aslonnie**
- add a script that can be used on buildkite pipeline to build and run rayci
- add logic to generate basic, noop buildkite pipeline
- self bootstraping on rayci's ci.

### CODE REVIEW

**Comment by aslonnie** [comment]
noop step generated: https://buildkite.com/ray-project/rayci/builds/98#018970ee-aa20-46e8-9a06-17fba8bb04c6

**Comment by aslonnie** [comment]
with this merged, we can create new civ2 pipelines with bootstrap commands.

File: raycicmd/main.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	}
+
+	// Upload pipeline to buildkite.
+	bs, err := yaml.Marshal(pipeline)
+	if err != nil {
+		return fmt.Errorf("marshal pipeline: %w", err)
+	}
+
+	if !flags.UploadPipeline {
+		if flags.OutputFile == "-" {
```
Comment (can-anyscale) [inline raycicmd/main.go:104]: what is this character -?

**Review by can-anyscale — APPROVED** [review]

File: raycicmd/make.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+func makePipeline(repoDir string, config *config, buildID string) (
+	*bkPipeline, error,
+) {
+	pl := new(bkPipeline)
+
+	// TODO(aslonnie): build rayci pipeline here.
+
+	totalSteps := 0
+	for _, g := range pl.Steps {
```
[the remark this answers is not in the document]
Comment (aslonnie) [reply raycicmd/make.go:11]: because each step is a group.

[the remark this answers is not in the document]
Comment (aslonnie) [reply raycicmd/make.go:11]: renamed to `group`

File: raycicmd/main.go (hunk 1 above)
Comment (aslonnie) [reply raycicmd/main.go:104]: means stdout. it is pretty common that `-` means stdin or stdout in flags or args.

**Pull request merged**